src/components/ResourceList/components/FilterControl/components/DateSelector/DateSelector.tsx

```diff
@@ -408,8 +411,22 @@ function getDateFilterOption(
   return filterValue;
 }
 
-function formatDateValue(date: Date) {
-  return date.toISOString().slice(0, 10);
+function formatDateForTimezone(date: Date) {
+  const timezoneOffset = date.getTimezoneOffset() * 60 * 1000;
+  if (date.getTimezoneOffset() > 0) {
+    const newTime = date.getTime() + timezoneOffset;
+    date.setTime(newTime);
```

Comment:
I think your function should not mutate the received date, but instead return a new `Date` object?